pkg/scheduler/framework/runtime/framework.go

```diff
@@ -594,6 +595,101 @@ func (f *frameworkImpl) runPostFilterPlugin(ctx context.Context, pl framework.Po
 	return r, s
 }
 
+// RunFilterPluginsWithNominatedPods runs the set of configured filter plugins
+// for nominated pod on the given node.
+func (f *frameworkImpl) RunFilterPluginsWithNominatedPods(ctx context.Context, state *framework.CycleState, pod *v1.Pod, info *framework.NodeInfo) (bool, *framework.Status, error) {
+	return PodPassesFiltersOnNode(ctx, f.PreemptHandle(), state, pod, info)
+}
+
+// PodPassesFiltersOnNode checks whether a node given by NodeInfo satisfies the
+// filter plugins.
+// This function is called from two different places: Schedule and Preempt.
+// When it is called from Schedule, we want to test whether the pod is
+// schedulable on the node with all the existing pods on the node plus higher
+// and equal priority pods nominated to run on the node.
+// When it is called from Preempt, we should remove the victims of preemption
+// and add the nominated pods. Removal of the victims is done by
+// SelectVictimsOnNode(). Preempt removes victims from PreFilter state and
+// NodeInfo before calling this function.
+func PodPassesFiltersOnNode(
```

Comment:
we don't need this separate function, RunFilterPluginsWithNominatedPods should be sufficient (i.e., put this logic inside RunFilterPluginsWithNominatedPods directly).